src/io/import/importDataSources.ts

```diff
@@ -58,72 +35,138 @@ const unhandledResource: ImportHandler = () => {
-    handleConfig,
+    // before extractArchive as .zip extension is part of state file check
     restoreStateFile,
     handleRemoteManifest,
     handleGoogleCloudStorage,
     handleAmazonS3,
     downloadUrl,
     extractArchiveTargetFromCache,
     extractArchive,
+    handleConfig, // collect to apply later
     // should be before importSingleFile, since DICOM is more specific
     handleDicomFile,
     importSingleFile,
     // catch any unhandled resource
     unhandledResource,
   ];
+  const loader = new Pipeline(middleware);
 
-  const importContext = {
-    fetchFileCache: new Map<string, File>(),
-    dicomDataSources: [] as DataSourceWithFile[],
-  };
-
-  const pipeline = new Pipeline(middleware);
   const results = await Promise.all(
-    dataSources.map((r) => pipeline.execute(r, importContext))
+    dataSources.map((r) => loader.execute(r, importContext))
   );
 
-  if (!importContext.dicomDataSources.length) {
-    return results;
-  }
+  // Apply config.JSONs
+  const configResult = await (async () => {
+    try {
+      results
+        .flatMap((pipelineResult) =>
+          pipelineResult.ok ? pipelineResult.data : []
+        )
+        .map((result) => result.config)
+        .filter(nonNullable)
+        .forEach(applyConfig);
+      return {
+        ok: true as const,
+        data: [],
+      };
+    } catch (err) {
+      return {
+        ok: false as const,
+        errors: [
+          {
+            message: toMeaningfulErrorString(err),
+            cause: err,
+            inputDataStackTrace: [],
+          },
+        ],
+      };
+    }
+  })();
 
   // handle DICOM loading
-
   const dicomDataSource: DataSource = {
     dicomSrc: {
       sources: importContext.dicomDataSources,
     },
   };
-  const dicomResult: PipelineResult<DataSource, ImportResult> = {
-    ok: true,
-    data: [],
-    errors: [],
-  };
+  const dicomResult = await (async () => {
+    try {
+      if (!importContext.dicomDataSources.length) {
+        return {
+          ok: true as const,
+          data: [],
+        };
+      }
+      const volumeKeys = await useDICOMStore().importFiles(
+        importContext.dicomDataSources
+      );
+      return {
+        ok: true as const,
+        data: volumeKeys.map(
+          (key) =>
+            ({
+              dataID: key,
+              dataType: 'dicom' as const,
+              dataSource: dicomDataSource,
+            } as const)
+        ),
+      };
+    } catch (err) {
+      return {
+        ok: false as const,
+        errors: [
+          {
+            message: toMeaningfulErrorString(err),
+            cause: err,
+            inputDataStackTrace: [dicomDataSource],
+          },
+        ],
+      };
+    }
+  })();
+
+  return [
+    ...results,
+    dicomResult,
+    configResult,
+    // remove ok results that have no result data
+  ].filter((result) => !result.ok || (result.ok && result.data.length));
```

Comment:
Simplification: `!result.ok || result.data.length`, since typescript will infer `result.ok` in the second branch.